readgroup.wdl

```diff
@@ -66,39 +64,34 @@ workflow Readgroup {
         String chunksR2 = "${readgroupDir}/chunk_${chunk}/${chunk}_2.fq.gz"
     }
 
-    call biopet.FastqSplitter as fastqsplitterR1 {
-        input:
-            inputFastq = copyR1.outputFile,
-            outputPaths = chunksR1,
-            dockerTag = dockerTags["biopet-fastqsplitter"]
-    }
-
-
-    if (defined(readgroup.reads.R2)){
-        call biopet.FastqSplitter as fastqsplitterR2 {
+    if (numberChunks > 1) {
+        call biopet.FastqSplitter as fastqsplitterR1 {
             input:
-                inputFastq = select_first([copyR2.outputFile]),
-                outputPaths = chunksR2,
+                inputFastq = copyR1.outputFile,
+                outputPaths = chunksR1,
                 dockerTag = dockerTags["biopet-fastqsplitter"]
         }
+        if (defined(readgroup.reads.R2)){
+            call biopet.FastqSplitter as fastqsplitterR2 {
+                input:
+                    inputFastq = select_first([copyR2.outputFile]),
+                    outputPaths = chunksR2,
+                    dockerTag = dockerTags["biopet-fastqsplitter"]
+            }
+        }
     }
 
-    scatter (x in range(length(chunksR1))) {
-        FastqPair chunks = if defined(fastqsplitterR2.chunks)
-            then {"R1": fastqsplitterR1.chunks[x],
-                "R2": select_first([fastqsplitterR2.chunks])[x]}
-            else {"R1": fastqsplitterR1.chunks[x]}
-    }
 
     # QC and Mapping
-    scatter (chunk in chunks) {
-        String chunkDir = sub(chunk.R1, basename(chunk.R1), "")
+    scatter (x in range(length(chunksR1))) {
+        File  chunk_read1 = if defined(fastqsplitterR1.chunks) then select_first([fastqsplitterR1.chunks])[x] else reads.R1
```

Comment:
Couldn't `select_all` be used here, instead of putting the array in an array first?

Reply:
No, because it is the array itself which is optional. Not the elements in it.